hoomd/md/external/wall.py

```diff
@@ -0,0 +1,653 @@
+        keys:
+
+        * ``epsilon`` (`float`, **required**) -
+          energy parameter :math:`\varepsilon` :math:`[\mathrm{energy}]`
+        * ``sigma`` (`float`, **required**) -
+          particle size :math:`\sigma` :math:`[\mathrm{length}]`
+        * ``r_cut`` (`float`, **required**) -
+          The cut off distance for the wall potential :math:`[\mathrm{length}]`
+        * ``r_extrap`` (`float`, **optional**) -
+          The distance to extrapolate the potential, defaults to 0
+          :math:`[\mathrm{length}]`
+
+        Type: `TypeParameter` [``particle_types``, `dict`]
+    """
+
+    _cpp_class_name = "WallsPotentialForceShiftedLJ"
+
+    def __init__(self, walls=None):
+
+        # initialize the base class
+        super().__init__(walls)
+
+        params = hoomd.data.typeparam.TypeParameter(
+            "params", "particle_types",
+            hoomd.data.parameterdicts.TypeParameterDict(epsilon=float,
+                                                        sigma=float,
+                                                        r_cut=float,
+                                                        r_extrap=0.0,
+                                                        len_keys=1))
+        self._add_typeparam(params)
+
+
+class Mie(WallPotential):
+    r"""Mie potential wall potential.
+
+    Args:
+        walls (`list` [`hoomd.wall.WallGeometry` ]): A list of wall definitions
+            to use for the potential.
+
+    Wall force evaluated using the Mie potential.  See `hoomd.md.pair.Mie` for
+    force details and base parameters and `WallPotential` for generalized wall
+    potential implementation
+
+    Example::
+
+        walls = [hoomd.wall.Sphere(radius=4.0)]
+        # add walls to interact with
+        mie_wall=hoomd.md.wall.Mie(walls, default_r_cut=3.0)
+        mie_wall.params['A'] = {
+            "epsilon": 1.0, "sigma": 1.0, "n": 12, "m": 6, "r_cut": 3.0}
+        mie_wall.params[['A','B']] = {
+            "epsilon": 0.5, "sigma": 3.0, "n": 49, "m": 50, "r_cut": 3.2}
+
+    Attributes:
+        walls (`list` [`hoomd.wall.WallGeometry` ]): A list of wall definitions
+            to use for the potential.
+
+    .. py:attribute:: params
+
+        The potential parameters per type. The dictionary has the following
+        keys:
+
+        * ``epsilon`` (`float`, **required**) -
+          energy parameter :math:`\varepsilon` :math:`[\mathrm{energy}]`
+        * ``sigma`` (`float`, **required**) -
+          particle size :math:`\sigma` :math:`[\mathrm{length}]`
+        * ``r_cut`` (`float`, **required**) -
+          The cut off distance for the wall potential :math:`[\mathrm{length}]`
+        * ``r_extrap`` (`float`, **optional**) -
+          The distance to extrapolate the potential, defaults to 0
+          :math:`[\mathrm{length}]`
+
+        Type: `TypeParameter` [``particle_types``, `dict`]
+    """
+
+    _cpp_class_name = "WallsPotentialMie"
+
+    def __init__(self, walls=None):
```

Comment:
```suggestion
    def __init__(self, walls):
```
What use-case would users have for initializing the class with no walls?

This comment applies to all the wall classes.

Reply:
To set the walls later, so they don't have to be defined at construction, but we can require it at construction.